File: app/src/main/java/com/kota/Bahamut/Pages/Model/BoardPageBlock.java
package com.kota.Bahamut.Pages.Model;

import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Bahamut.Pages.BoardPage.BoardPageAction;

import java.util.Stack;

public class BoardPageBlock extends TelnetListPageBlock {
    private static final Stack<BoardPageBlock> _pool = new Stack<>();
    public String BoardManager = null;
    public String BoardName = null;
    public String BoardTitle = null;
    public int Type = BoardPageAction.LIST;
    public int mode = 0;

    private BoardPageBlock() {
    }

    public static void release() {
        synchronized (_pool) {
            _pool.clear();
        }
    }

    public static void recycle(BoardPageBlock block) {
        synchronized (_pool) {
            _pool.push(block);
        }
    }

    public static BoardPageBlock create() {
        BoardPageBlock block = null;
        synchronized (_pool) {
            if (_pool.size() > 0) {
                block = _pool.pop();
            }
        }
        if (block == null) {
            return new BoardPageBlock();
        }
        return block;
    }
}

File: app/src/main/java/com/kota/Bahamut/pages/model/BoardPageBlock.kt
package com.kota.Bahamut.pages.model

import com.kota.Bahamut.listPage.TelnetListPageBlock
import com.kota.Bahamut.pages.boardPage.BoardPageAction
import java.util.Stack

class BoardPageBlock private constructor() : TelnetListPageBlock() {
    var boardManager: String = ""
    var boardName: String = ""
    var boardTitle: String = ""
    var boardType: Int = BoardPageAction.LIST
    var boardMode: Int = 0

    companion object {
        private val _pool = Stack<BoardPageBlock>()
        @JvmStatic
        fun release() {
            synchronized(_pool) {
                _pool.clear()
            }
        }

        /** 回收區塊物件，清空看板相關資訊與內部項目以釋放記憶體引用 */
        fun recycle(block: BoardPageBlock) {
            block.clear() // 清除基類 Items 陣列引用
            // 清除本類別欄位
            block.boardManager = ""
            block.boardName = ""
            block.boardTitle = ""
            block.boardType = BoardPageAction.LIST
            block.boardMode = 0
            synchronized(_pool) {
                _pool.push(block)
            }
        }

        fun create(): BoardPageBlock {
            var block: BoardPageBlock? = null
            synchronized(_pool) {
                if (_pool.isNotEmpty()) {
                    block = _pool.pop()
                }
            }
            if (block == null) {
                return BoardPageBlock()
            }
            return block
        }
    }
}
